File: 2023/Exercicios/Set e Busca Binaria/1215.py
palavras = set()
alfabeto = "abcdefghijklmnopqrstuvwxyz"
while True:
    try:
        palavra_atual = ""
        s = input().lower()
        print(s)
        for i in s:
            if i in alfabeto:
                palavra_atual += i
            else:
                if palavra_atual != "":
                    palavras.add(palavra_atual)
                palavra_atual = ""
        if palavra_atual != "":
            palavras.add(palavra_atual)
    except EOFError:
        for i in palavras:
            print(i)
        break

File: 2024/programacao_avancada/2024.1/26-04-2024/1215.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    string s, palavra_atual;
    set<string> palavras;
    string alfabeto = "abcdefghijklmnopqrstuvwxyz";
    while (cin >> s){
        stringstream ss(s);
        string word;
        while (ss >> palavra_atual){
            for (auto x: palavra_atual){
                if (alfabeto.find(tolower(x)) != string::npos && x != ' ' && x != '\n'){
                    word += tolower(x);
                } else {
                    if (word != "") palavras.insert(word);
                    word = "";
                }
            }
            if (word != ""){
                palavras.insert(word);
            }   

        }
    }
    for (auto x: palavras){
        cout << x << "\n";
    }
    return 0;
}